src/color_picker.coffee

```diff
@@ -210,16 +210,13 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       ])
     ])
 
-  setupCustomColor: Ember.on 'init', ->
-    selectedColor = colorToHex(@get('selectedColor'))
-    return if @get('colorRows').find (row) ->
+  setCustomColor: Ember.observer(->
+    selectedColor = @get 'selectedColor'
+    selectedColor = colorToHex(selectedColor)
+    return @set('customColor', '') if @get('colorRows').find (row) ->
       selectedColor in row.invoke 'toLowerCase'
-    @set 'customColor', @get('selectedColor')
-
-  setCustomColor: Ember.observer ->
-    if @get('isCustomColorValid') is true
-      @set 'selectedColor', expandHexColor(@get('customColor'))
-  , 'customColor', 'isCustomColorValid'
+    @set 'customColor', selectedColor
+  , 'selectedColor', 'colorRows').on('init')
```

Comment:
I always do the mistake to think only about `function` when talking about prototype extensions, but it has some other implications ( See  http://guides.emberjs.com/v1.10.0/configuring-ember/disabling-prototype-extensions ). We should be carefult about:

> If you disable prototype extension and attempt to use native arrays with things like a template's {{#each}} helper, Ember.js will have no way to detect changes to the array and the template will not update as the underlying array changes.